cluster.yml

```diff
@@ -55,8 +55,8 @@
 - hosts: k8s-cluster
   any_errors_fatal: true
   roles:
-    - { role: dnsmasq, tags: dnsmasq }
-    - { role: kubernetes/preinstall, tags: resolvconf }
+    - { role: dnsmasq, when: "dns_mode == 'dnsmasq_kubedns'", tags: dnsmasq }
```

Comment:
This was unconditional before and had the effect of installing dnsmasq even if skip_dnsmasq was true.

I now changed the variables to:
```yaml
# Can be dnsmasq_kubedns, kubedns or none
dns_mode: dnsmasq_kubedns

# Can be docker_dns, host_resolvconf or none
hostnet_dns_mode: host_resolvconf
```

This should give us the most flexibility.